tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -128,7 +135,9 @@ object NativeConfig {
       linkStubs = false,
       optimize = true,
       linktimeProperties = Map.empty,
-      embedResources = false
+      embedResources = false,
+      nativeSourcePlugins =
+        Seq(CSourcePlugin, CppSourcePlugin, RustSourcePlugin)
```

Comment:
I put them all here for now, for backwards-compatibility. Maybe in future it should be empty, and default C/CPP plugins are populated elsewhere?

Reply:
You made another comment:
> Maybe, if in the future someone would like to introduce better Cpp integration we could add settings to nativeConfig allowing for disabling our internal implementations.

I think instead of adding more settings, we should keep the C and Cpp plugins as part of the configurable list instead of force-enabling them. They can be included by default but then they will be easy to remove/replace. They should definitely remain part of the core repo always :)

I think the Rust makes sense as another repo.